### PR TITLE
[MM-60789] Replace FormattedMarkdownMessage in 'webapp/channels/src/components/channel_selector_modal/channel_selector_modal.tsx' with FormattedMessage

### LINKED ISSUE #28447: Replace FormattedMarkdownMessage in 'webapp/channels/src/components/channel_selector_modal/channel_selector_modal.tsx'
The goal of this task is to remove all instances of  “FormattedMarkdownMessage” (imported from  “components/formatted_markdown_message”) in the file  `webapp/channels/src/components/channel_selector_modal/channel_selector_modal.tsx`.  These should be replaced with the “FormattedMessage” component from the  “react-intl” library.

Ensure that rich text formatting is preserved by following the implementation guidelines in the react-intl ([https://formatjs.io/docs/react-intl/components#rich-text-formatting](https://formatjs.io/docs/react-intl/components#rich-text-formatting))  documentation. Make sure to change the message ID after the update as the current extraction tool does not automatically detect changes in the messages. Once done, make sure to run `npm run i18n-extract` in the “webapp/channels” directory to pick up the new messages. For guidance refer to this example pr - [https://github.com/mattermost/mattermost/pull/28438](https://github.com/mattermost/mattermost/pull/28438)

----
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-60789

### PR DESCRIPTION
**Pull request by RS-labhub**
#### Summary
Replaced FormattedMarkdownMessage in '../channel_selector_modal/channel_selector_modal.tsx' to FormattedMessage

#### Ticket Link

Fixes #28447 

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by RS-labhub** [comment]
This is something fishy! 

`npm run i18n-extract` this test didn't throw any error, but pr-ci failed!

I changed `FormattedMarkdownMessage` to `FormattedMessage` but the CI showing `memoizedFormattedMessage`

**Comment by M-ZubairAhmed** [comment]
> This is something fishy!
> 
> `npm run i18n-extract` this test didn't throw any error, but pr-ci failed!
> 
> I changed `FormattedMarkdownMessage` to `FormattedMessage` but the CI showing `memoizedFormattedMessage`

Can you also `$ npm run test:updatesnapshot` in the webapp directory that will update the snapshots

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]
Please check the description on what needs to be done https://github.com/mattermost/mattermost/issues/28447#issue-2564147979

**Comment by RS-labhub** [comment]
Checked it! There are no links there, so I guess that's not troubling...

I guess `$ npm run test:updatesnapshot` this should work.  

And I see, you are also using `MemoizedFormattedMessage` rather than `FormattedMessage` as given in the description. Refer the image:
image

**Comment by M-ZubairAhmed** [comment]
> Checked it @M-ZubairAhmed ! There are no links there, so I guess that's not troubling...
> 
> I guess `$ npm run test:updatesnapshot` this should work.
> 
> And I see, you are also using `MemoizedFormattedMessage` rather than `FormattedMessage` as given in the description. Refer the image: image

You are looking at the snapshot. You should be looking at the tsx file rather

**Comment by RS-labhub** [comment]
Ehehe.. Will check it tomorrow! thanks for helping!

**Comment by RS-labhub** [comment]
@M-ZubairAhmed could it review it now?? If everything will be good, I'll squash the commits into one!

File: webapp/channels/src/components/channel_selector_modal/channel_selector_modal.tsx [hunk 1]
```diff
@@ -236,9 +235,12 @@ export class ChannelSelectorModal extends React.PureComponent<Props, State> {
                         componentClass='h1'
                         id='channelSelectorModalLabel'
                     >
-                        <FormattedMarkdownMessage
-                            id='add_channels_to_scheme.title'
-                            defaultMessage='Add Channels to **Channel Selection** List'
+                        <FormattedMessage
+                            id='manage_scheme_channels.title'
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/channel_selector_modal/channel_selector_modal.tsx:239]: Nit: let's change the id to "channelSelectorModal.title", you have to delete the translation in en file first and run the extract command again

File: webapp/channels/src/components/channel_selector_modal/channel_selector_modal.tsx [hunk 2]
```diff
@@ -4,13 +4,12 @@
 import React from 'react';
 import {Modal} from 'react-bootstrap';
 import type {IntlShape} from 'react-intl';
-import {injectIntl, FormattedMessage} from 'react-intl';
+import {FormattedMessage, injectIntl} from 'react-intl';
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/channel_selector_modal/channel_selector_modal.tsx:7]: Why did we change the order here?

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]
Looks perfect just some small nits

File: webapp/channels/src/components/channel_selector_modal/channel_selector_modal.tsx (hunk 2 above)
Comment (RS-labhub) [reply webapp/channels/src/components/channel_selector_modal/channel_selector_modal.tsx:7]: image

I thought that's the pattern you're following.. Loll, it's not same everywhere!

File: webapp/channels/src/components/channel_selector_modal/channel_selector_modal.tsx (hunk 1 above)
Comment (RS-labhub) [reply webapp/channels/src/components/channel_selector_modal/channel_selector_modal.tsx]: Done!

File: webapp/channels/src/components/channel_selector_modal/channel_selector_modal.tsx (hunk 2 above)
Comment (RS-labhub) [reply webapp/channels/src/components/channel_selector_modal/channel_selector_modal.tsx:7]: Let me know if I have to reverse it, after this I'll squash all the commits into one!

Comment (M-ZubairAhmed) [reply webapp/channels/src/components/channel_selector_modal/channel_selector_modal.tsx:7]: Yes we recommend not introducing any unintended changes and keeping the scope to what's required in the PR itself. Not saying this is a big change but it's a good practice to follow

Comment (RS-labhub) [reply webapp/channels/src/components/channel_selector_modal/channel_selector_modal.tsx]: Done!!

**Comment by RS-labhub** [comment]
Again failed!! Running `npm run test:updatesnapshot` again 💀

**Comment by RS-labhub** [comment]
image

I think I missed out something or did some sin 😭

**Comment by M-ZubairAhmed** [comment]
> Again failed!! Running `npm run test:updatesnapshot` again 💀
> 
> image
> 
> I think I missed out something or did some sin 😭

We also have a command to update the jest snapshots `npm run test:updatesnapshot`

**Comment by M-ZubairAhmed** [comment]
> Again failed!! Running `npm run test:updatesnapshot` again 💀
> 
> image
> 
> I think I missed out something or did some sin 😭

You just need to run the update snapshots again

**Comment by RS-labhub** [comment]
> You just need to run the update snapshots again

This time I'll not get errors?

**Comment by M-ZubairAhmed** [comment]
> > You just need to run the update snapshots again
> 
> This time I'll not get errors?
> 
> Edit: Some tests are still failing! @M-ZubairAhmed

can you push the changes? We can check in the CI

**Comment by RS-labhub** [comment]
> > > You just need to run the update snapshots again
> > 
> > 
> > This time I'll not get errors?
> > Edit: Some tests are still failing! @M-ZubairAhmed
> 
> can you push the changes? We can check in the CI

image

just a few more to execute!

**Comment by M-ZubairAhmed** [comment]
> > > > You just need to run the update snapshots again
> > > 
> > > 
> > > This time I'll not get errors?
> > > Edit: Some tests are still failing! @M-ZubairAhmed
> > 
> > 
> > can you push the changes? We can check in the CI
> 
> image
> 
> just a few more to be executed!

In CI just 2 failed snapshots are shows, it looks like there is a difference between your local test run and CI run.

**Comment by RS-labhub** [comment]
> > > > > You just need to run the update snapshots again
> > > > 
> > > > 
> > > > This time I'll not get errors?
> > > > Edit: Some tests are still failing! @M-ZubairAhmed
> > > 
> > > 
> > > can you push the changes? We can check in the CI
> > 
> > 
> > image
> > just a few more to be executed!
> 
> In CI just 2 failed snapshots are shows, it looks like there is a difference between your local test run and CI run.

image
this is the final result!

Do I need to worry? I'm pushing my changes!

**Comment by RS-labhub** [comment]
Ehhh.. it worked.. Ehehehe.. The tests were failing for other files...

**Comment by M-ZubairAhmed** [comment]
> Ehhh.. it worked.. Ehehehe.. The tests were failing for other files...

Its best to check your local setup if you have setup stuff correctly like npm etc

**Review by M-ZubairAhmed — APPROVED** [review]
This looks good :shipit: Thank you

**Review by Rajat-Dabade — APPROVED** [review]
LGTM 🚀

**Review by yasserfaraazkhan — APPROVED** [review]

**Pull request merged**